### PR TITLE
Add check if canonical id is already in the system

### PR DESCRIPTION
**Pull request by matzew**

### CODE REVIEW

**Comment by matzew** [comment]
@sinarz I added an extra check to ypur branch. 

feel free to merge this one

**Comment by sinarezaei** [comment]
Thank you, a really good commit since it stops possible bugs from GCM or wrong/duplicate registrations done by the client. And if the previous token remains in the aerogear client installation it will be removed by the next push notification sent to GCM.

File: push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/GCMPushNotificationSender.java [hunk 1]
```diff
@@ -172,20 +172,25 @@ private void cleanupInvalidRegistrationIDsForVariant(String variantID, Multicast
...
-                //if the result contains a canonical id, there is no need to remove the current inactive token from data store
-                
-            }
-            else
-            {
+
+                // let's see if the canonical id is already in our system:
+                Installation installation = clientInstallationService.findInstallationForVariantByDeviceToken(variantID, canonicalRegId);
+
+                // there is device on this variant with the given canonical id
```
Comment (secondsun) [inline push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/GCMPushNotificationSender.java]: do you mean "there is not a device" ?

**Comment by secondsun** [comment]
@matzew Other than the typo looks good.

**Comment by matzew** [comment]
@secondsun updated 

@sinarz can you merge? Than I will be merging your PR against AG tomorrow :tada:

**Comment by matzew** [comment]
@sinarz @secondsun I added another if, to remove the old/duplicated token in case the canonical ID is already registered

**Comment by matzew** [comment]
closing, code has been rebased and squashed on https://github.com/aerogear/aerogear-unifiedpush-server/pull/693

**Pull request closed without merging**